content/articles/little-printer.md

```diff
@@ -0,0 +1,75 @@
+Title: Berg's Little Printer
+Date: 2015-04-30
+Category: Misc
+Authors: Tom
+Short_summary: We did some pretty exciting work for Berg!
+
+I'm a big fan of the amazing London design agency Berg. So when its
+CEO Matt Berg tweeted this last year:
+
+<blockquote class="twitter-tweet" lang="en"><p lang="en" dir="ltr">I&#39;m after a London Python freelancer for a 2mo gig, Flask/tests/Redis/Bootstrap. Mail me... matt at interconnected dot org. Tell yr friends!</p>&mdash; Matt Webb (@genmon) <a href="https://twitter.com/genmon/status/537304016993386497">November 25, 2014</a></blockquote> <script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+I was hooked.
+
+<!-- PELICAN_END_SUMMARY -->
+
+We emailed Matt and it turned out that he was looking for someone to
+re-implement a server for their
+[Little Printer](http://www.wired.co.uk/news/archive/2011-11/29/berg-little-printer-bergcloud).
+
+Along with the sad news that Berg was
+[shutting down](http://blog.bergcloud.com/2014/09/09/week-483/) came
+the good news that Matt wanted the Little Printer to live on. And the
+best way to do so was to have a simpler server, divorced from the rest
+of Berg's offerings.
+
+By the time we arrived Matt had already implemented a few fairly
+complicated parts for encoding the binary data sent to the Little
+Printer. Despite that there were plenty of challenges left for us.
+
+
+## Challenges
+
+* All printers are constantly connected to the server, and need to
+  react quickly to push notifications.
+* The server should be easy to deploy: We needed to avoid complicated
+  dependencies and interactions between moving parts.
+* The protocol wasn't documented other than in the existing code base.
+
+
+## Protocol implementation
+
+We chose to go with a simple stratified design: We have a
+socket-handling layer that deals with the online/offline status for
+each printer. On top of that we have a message encoding and decoding
+layer that translates the payload into Python's
+[namedtuples](https://docs.python.org/2/library/collections.html#collections.namedtuple). The
+last layer is a small website for user interactions.
```

Comment:
I would put an example of namedtuple since it's pretty unknown despite being cool, we don't people to leave the article if possible and more people seeing namedtuple is a good thing !